### ISSUE TITLE
ZIM Metadata are checked too late

### ISSUE THREAD

**Issue by kelson42** [openzim/zimit#237]
Currently, and this is only an example, if the ZIM Metadata `Description` is over 80 characters, then the scrape dies at the end of the process (after the Web site has been spidered).

It should be checked and handled properly (if not correct) at the very beginning of the scrape.

This include both cases:
* Metadata is given on the command line
* Metadata is retrieved from the scrape

This remark is valid for [all ZIM metadata](https://wiki.openzim.org/wiki/Metadata).

**Comment by rgaudin** [comment]
> Metadata is retrieved from the scrape

This is not really possible unless we decide that we (ie. not the scraper) find, parse, fetch and process those on our own.

**Comment by benoit74** [comment]
In current context (zimit/warc2zim), which ZIM metadata is retrieved from the scrape? (I checked and found none, but I may have read the code too fast / missed something).

First point about metadata given on the command line is still valid.

**Comment by rgaudin** [comment]
> I checked and found none, but I may have read the code too fast / missed something

It's in warc2zim:

- Language
- Illustration
- Source (set to the mainUrl if not provided)

**Comment by benoit74** [comment]
So this is an important question is as @rgaudin said: do we decide:
- to fetch these three informations on our own (i.e. not rely on browsertrix crawler)
- to fallback nicely / provide only correct value

From my PoV, we should implement the second option, i.e. ensure these 3 parameters are always valid, this does not look like an impossible task and would avoid to have to maintain code doing stuff already done by the crawler (e.g. fetching the illustration is known to be a complex task AFAIK, so I would prefer to ensure the illustration retrieved by the scraper is valid, and if not provide a fallback image so that the ZIM is created in all situations)

**Comment by benoit74** [comment]
Metadata which are configured by the user on the command line (e.g. description) are now checked at scraper startup. 

Other metadata (language, illustration, source) are available only once the scrape is done for obvious reasons.

Do we want to alter zimit so that it calls the crawler twice to fetch and check metadata asap?
  - call crawler with proper settings so that it fetches only the homepage
  - retrieve and check metadata from WARC file
  - call crawler with full settings and build the ZIM as of today

Is there any alternative? 

I consider this is a nice to have which does not needs to be solved in 2.0 but can wait for later
